app/main.py

```diff
@@ -1,3 +1,23 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Hashable, ParamSpec, TypeVar
+
+Param = ParamSpec("Param")
+RetType = TypeVar("RetType")
+
+
+def cache(func: Callable[Param, RetType]) -> Callable[Param, RetType]:
+    cache_dict: dict = {}
+
+    def wrapper(*args: Param.args, **kwargs: Param.kwargs) -> RetType:
+        all_arguments = args + tuple(kwargs.values())
+        for arg in all_arguments:
+            if not isinstance(arg, Hashable):
+                return func(*args, **kwargs)
+        signature: tuple = (args, tuple(kwargs.items()))
```

Comment:
I believe this approach doesn't work for such calls:

```
def func(a, b):
  ...

func(1, 2)
func(a=1, b=2)
```
signature are different for those two cases

let's assume that only positional arguments are used